packages/db/src/repositories/boost.ts

```diff
@@ -689,25 +689,25 @@ class BoostRepository {
   ): Promise<BoostAgentResult> {
     const executor = tx || this.#db;
     return await executor.transaction(async (tx) => {
-      // 1) Lock the agent boost total row if it exists
-      const [agentBoostTotal] = await tx
-        .select()
-        .from(schema.agentBoostTotals)
-        .where(
-          and(
-            eq(schema.agentBoostTotals.agentId, agentId),
-            eq(schema.agentBoostTotals.competitionId, competitionId),
-          ),
-        )
-        .limit(1)
-        .for("update"); // locks this row until tx ends
```

Comment:
Based on the Supabase query performance advisor, this lock is slowing down the database.  This query has been averaging over 2 seconds per query, and in one case it took 23 seconds.  

More importantly, we don't use this value unless the `diffRes.type === "noop"`, so we don't need to do the query on every method call.